src/test/java/com/thomashan/coup/PlayerCardsTest.java

```diff
@@ -0,0 +1,105 @@
+package com.thomashan.coup;
+
+import com.thomashan.collection.immutable.ImmutableList;
+import org.junit.jupiter.api.Test;
+
+import static com.thomashan.coup.Card.AMBASSADOR;
+import static com.thomashan.coup.Card.ASSASSIN;
+import static com.thomashan.coup.Card.CAPTAIN;
+import static com.thomashan.coup.Card.DUKE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+import static org.junit.jupiter.api.Assertions.assertNotSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class PlayerCardsTest {
+    @Test
+    public void testOf_ReturnPlayerCards() {
+        assertEquals(PlayerCards.class, PlayerCards.of(AMBASSADOR, ASSASSIN).getClass());
+    }
+
+    @Test
+    public void testGetActivePlayerCards_GivenSameCard_ReturnSingleElement() {
+        assertEquals(1, PlayerCards.of(AMBASSADOR, AMBASSADOR).getActivePlayerCards().size());
+    }
+
+    @Test
+    public void testGetActivePlayerCards_GivenDifferentCards_ReturnAllElement() {
+        assertEquals(2, PlayerCards.of(AMBASSADOR, ASSASSIN).getActivePlayerCards().size());
+    }
+
+    @Test
+    public void testGet() {
+        assertIterableEquals(ImmutableList.of(PlayerCard.of(AMBASSADOR), PlayerCard.of(AMBASSADOR)), PlayerCards.of(AMBASSADOR, AMBASSADOR).get());
+    }
+
+    @Test
+    public void testReveal_GivenPlayerDoesNotHaveTheCard_ThrowsAnException() {
+        Throwable throwable = assertThrows(IllegalArgumentException.class, () -> PlayerCards.of(AMBASSADOR, AMBASSADOR).reveal(PlayerCard.of(ASSASSIN)));
+        assertEquals("Player does not have the card", throwable.getMessage());
+    }
+
+    @Test
+    public void testReveal_GivenPlayerHasTheCardButDifferentRevealState_ThrowsAnException() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unit tests should not contain more than 1 assert(s).](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4561385)